pkg/cli/admin/sosreport/sosreport.go

```diff
@@ -0,0 +1,687 @@
+package sosreport
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+	"math/rand"
+	"os"
+	"path"
+	"regexp"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/spf13/cobra"
+	"k8s.io/klog/v2"
+
+	configclient "github.com/openshift/client-go/config/clientset/versioned"
+	imagev1client "github.com/openshift/client-go/image/clientset/versioned/typed/image/v1"
+	imagereference "github.com/openshift/library-go/pkg/image/reference"
+	"github.com/openshift/library-go/pkg/operator/resource/retry"
+	"github.com/openshift/oc/pkg/cli/admin/inspect"
+	"github.com/openshift/oc/pkg/cli/rsync"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/cli-runtime/pkg/printers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/kubectl/pkg/cmd/logs"
+	kcmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/polymorphichelpers"
+	"k8s.io/kubectl/pkg/scheme"
+	"k8s.io/kubectl/pkg/util/templates"
+)
+
+var (
+	sosReportLong = templates.LongDesc(`
+		Launch a pod to collect a SOSReport from the provided node-name
+
+		This command will launch a pod in a temporary namespace on your cluster that gathers
+		a SOSReport from the provided node and then downloads the collected report.
+
+		Experimental: This command is under active development and may change without notice.
+	`)
+
+	sosReportExample = templates.Examples(`
+		# Collect a SOSReport using the default image configuration, writing the SOSReport into ./<host-name>.sosreport
+		  oc adm sos
+	`)
+)
+
+func NewSOSReportCommand(f kcmdutil.Factory, streams genericclioptions.IOStreams) *cobra.Command {
+	o := NewSOSReportOptions(streams)
+	cmd := &cobra.Command{
+		Use:     "sos",
+		Short:   "Launch a new instance of a pod for collecting a SOSReport for a specific node.",
+		Long:    sosReportLong,
+		Example: sosReportExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			kcmdutil.CheckErr(o.Complete(f, cmd, args))
+			kcmdutil.CheckErr(o.Validate())
+			kcmdutil.CheckErr(o.Run())
+		},
+	}
+
+	cmd.Flags().StringVar(&o.NodeName, "node-name", o.NodeName, "Set a specific node to use - by default a random master will be used")
+	cmd.Flags().StringSliceVar(&o.Images, "image", o.Images, "Specify a must-gather plugin image to run. If not specified, OpenShift's default must-gather image will be used.")
```

Comment:
Why would we use a must-gather image? Why not just use the RHEL tools image? 

cc @control-d @TurboTurtle

Reply:
It'd be `support-tools`, but yes I'd agree here to use an existing image. Additionally, the `support-tools` image contains the `RUN` label which defines how to launch the container so that `sos` collects from the node and not the container it is running in.